### ISSUE TITLE
assert.isNotNumber(NaN) does not pass. I consider it a bug

### ISSUE THREAD

**Issue by andrerpena** [chaijs/chai#477]
Hi.
I have number manipulation methods that are supposed to return NaN when incorrect input is passed in.
While checking `assert.isNotNumber(NaN)`, chai triggers:

    AssertionError: expected NaN not to be a number
	    at Function.assert.isNotNumber (base/test/index.js?af4a3dfe4a417464ed76a60bea36abe7aa63af34:6661:40)
	    at Context.<anonymous> (base/test/index.js?af4a3dfe4a417464ed76a60bea36abe7aa63af34:79:17)

I consider it to be a bug. Is that intentional?

**Comment by keithamus** [comment]
Hey @andrerpena thanks for the issue.

`assert.isNotNumber()` is the inverse of `assert.isNumber`. It does a check to see if the variable passed is of a type of Number. [See the code here](https://github.com/chaijs/chai/blob/master/lib/chai/interface/assert.js#L533-L549) - it effectively does `typeof obj !== 'number'`. In other words, the assertion is designed to test the number type, not to test if the value passes is `NaN` - the naming is very close, so I can understand the confusion.

If you're trying to assert whether or not the value passes is `NaN` it is best to use another assertion. I'd suggest one of the following:

```js
assert.deepEqual(NaN, NaN, 'value should be NaN');
assert(Number.isNaN(NaN), 'value should be NaN')
```

I think we are missing an `assert.isNaN` function. I'd happily accept a PR adding this - but I'd expect to see `assert.isNotNaN()`, `expect(v).to.be.NaN`, and `v.should.be.NaN` as well. Thoughts?

**Comment by andrerpena** [comment]
@keithamus , I'm sorry for the late response.
Thank you very much for the detailed explaination. That was helpful and it makes sense.

**Comment by keithamus** [comment]
@andrerpena how do you fancy making a pull request to add in `expect(v).to.be.NaN`?

**Comment by bradcypert** [comment]
We should be good to close this out now :beers: :white_check_mark:

**Comment by andrerpena** [comment]
@keithamus I'll try to implement it. I just can't confirm the date now. But I'll try.
@bradcypert , yes.. I'm closing it.

**Comment by bradcypert** [comment]
@andrerpena Just the clarify, I implemented this for the 3.1.0 build. If you just want to use it, consider pulling in the latest version of Chai. That being said, I fully encourage coding things yourself too! It's an excellent learning experience.
